src/core/shape/2d_primitives.js

```diff
@@ -560,13 +560,18 @@ p5.prototype.rect = function() {
   p5._validateParameters('rect', arguments);
```

Comment:
Maybe will be easier to understanding code if rewrite this function with default parameters, f.e.
```js
p5.prototype.rect = function(x, y, w, h = w, ...rest) {
  p5._validateParameters('rect', [x, y, w, h, ...rest]);

  if (this._renderer._doStroke || this._renderer._doFill) {
    const vals = Object.values(canvas.modeAdjust(x, y, w, h, this._renderer._rectMode));
    this._renderer.rect([...vals, ...rest]);
  }
  return this;
}
```

What do you think about this?

Reply:
I wonder if the `rect()` function is older than the `ellipse()` or written by someone else? Since that seems to be the function style used in `ellipse()`. Can't say I have an opinion either way – haha.. just wanted to allow my students to change `ellipse()` to `rect()` with the shape dissapearing.